felix/wireguard/bootstrap.go

```diff
@@ -185,22 +184,29 @@ func BootstrapHostConnectivityAndFilterTyphaAddresses(
 	return typhas, nil
 }
 
-// RemoveWireguardForHostEncryptionBootstrapping removes all wireguard configuration. This includes:
+// RemoveWireguardConditionallyOnBootstrap removes all wireguard configuration based on
+// configuration conditions. This includes:
 // - The wireguard public key
 // - The wireguard device (which in turn will delete all wireguard routing rules).
-func RemoveWireguardForHostEncryptionBootstrapping(
+func RemoveWireguardConditionallyOnBootstrap(
 	configParams *config.Config,
 	getNetlinkHandle func() (netlinkshim.Interface, error),
 	calicoClient clientv3.Interface,
 ) error {
-	if !configParams.WireguardHostEncryptionEnabled {
-		// HostEncryption is currently enabled in environments by operator rather than through FelixConfiguration.
-		// This should not change for a given deployment. Only host encryption should impact typha connectivity.
+	/*
+		| WireguardEnabled | WireguardHostEncryptionEnabled | Clear Wireguard PK + Device? |
+		|------------------|--------------------------------|------------------------------|
+		| YES			   | NO								| NO						   |
```

Comment:
Tabs vs spaces?  Note to self left in?

Reply:
We shouldn't need to handle the wireguard disabled case here since we handle that in the Bootstrap call above.  I updated daemon yesterday to always call into the BootstrapHostConnectivityAndFilterTyphaAddresses method even if wireguard is not enabled.  If not enabled then we'll remove wireguard config in there.

Reply:
This "always remove" wireguard is a last resort for host encryption only.

Reply:
Also, this table does not match the code.